handler/body.go

```diff
@@ -45,12 +45,12 @@ func (b wasmBody) Read(bytes []byte) (size uint32, eof bool) {
 		return // invalid, but prevent crashing.
 	}
 
-	size, eof = read(b, uintptr(ptr), limit)
+	size, eof = read(b, ptr, limit)
 	runtime.KeepAlive(bytes) // keep bytes alive until ptr is no longer needed.
 	return
 }
 
-func read(b wasmBody, ptr uintptr, limit imports.BufLimit) (size uint32, eof bool) {
+func read(b wasmBody, ptr uint32, limit imports.BufLimit) (size uint32, eof bool) {
```

Comment:
`uint32` looks fine but just curious did something change in 0.28 requiring that?

Reply:
`//go:wasmimport` is more restrictive on types accepted, particularly normal Go.

Reply:
here's the tinygo one.. fishing for go.. https://github.com/tinygo-org/tinygo/blob/v0.28.1/compiler/symbol.go#L389-L393

Reply:
Thanks - no worries, glad I got to see it here before crashing into the wall :)

Reply:
hmm not sure this is it as it seems to support unsafe.Pointer? https://github.com/golang/go/blob/ee46f0b5084461978432aa20df003ac52500b0f0/src/cmd/compile/internal/ssagen/abi.go#L369-L390

@achille-roussel do you know the canonical restricted list allowed for imports? I'd like to refer back to this issue if I am curious again later ;)